cpp/include/cudf/types.h

```diff
@@ -165,6 +165,10 @@ typedef struct gdf_context_{
   int flag_distinct;            /**< for COUNT: DISTINCT = 1, else = 0 */
   int flag_sort_result;         /**< When method is GDF_HASH, 0 = result is not sorted, 1 = result is sorted */
   int flag_sort_inplace;        /**< 0 = No sort in place allowed, 1 = else */
+  int flag_groupby_include_nulls; /**< 0 = Nulls are ignored in group by keys (Pandas style), 
```

Comment:
1. Boolean flags should be `bool`s. 
2. No magic numbers in comments. If it were a bool flag, with a proper name, then even without a comment it would be relatively clear what it means.
3. What does it mean that "NULLs are ignored by group by keys" - does that mean the tuples with NULLs are dropped? Not clear.
4. Instead of just saying what the pandas-style is, there should be some sort of mechanism for saying "handle nulls pandas-style, e.g. some kind of struct/class which, when applied to a gdf_context, sets this flag to have its pandas-style value.

Reply:
You could define your own `bool` as a `typedef enum`.

Reply:
@jrhemstad : C has had bools for nearly 20 years, actually. See: http://www.c-faq.com/bool/booltype.html

Reply:
It's not native to the language. `types.h` cannot have any `include` statements because CFFI does not have a preprocessor.

Reply:
@jrhemstad : The native-to-the-language version is `_Bool`. Of course, when including in C++ we would need to prepend `using _Bool = bool`;

but it's a bit baroque, so never mind.